application/services/Evaluation.php

```diff
@@ -1371,7 +1365,7 @@ public function getSummaryReportsInPdf($shipmentId) {
                         'rif_indeterminate' => new Zend_Db_Expr("SUM(CASE WHEN `res`.`rif_resistance` = 'indeterminate' THEN 1 ELSE 0 END)"),
                         'rif_uninterpretable' => new Zend_Db_Expr("SUM(CASE WHEN IFNULL(`res`.`mtb_detected`, '') IN ('noResult', 'invalid', 'error', '') THEN 1 ELSE 0 END)"),
                         'no_of_responses' => new Zend_Db_Expr('COUNT(*)'),
-                        'average_ct' => new Zend_Db_Expr('SUM(CASE WHEN IFNULL(`res`.`calculated_score`, \'pass\') <> \'fail\' THEN IFNULL(`res`.`probe_2`, 0) ELSE 0 END) / SUM(CASE WHEN `res`.`probe_2` IS NULL OR IFNULL(`res`.`calculated_score`, \'pass\') = \'fail\' THEN 0 ELSE 1 END)')))
+                        'average_ct' => new Zend_Db_Expr('CASE WHEN (SUM(CASE WHEN IFNULL(`res`.`calculated_score`, \'pass\') <> \'fail\' THEN IFNULL(`res`.`probe_2`, 0) ELSE 0 END) / SUM(CASE WHEN `res`.`probe_2` IS NULL OR IFNULL(`res`.`calculated_score`, \'pass\') = \'fail\' THEN 0 ELSE 1 END))<1 THEN 0 ELSE SUM(CASE WHEN IFNULL(`res`.`calculated_score`, \'pass\') <> \'fail\' THEN IFNULL(`res`.`probe_2`, 0) ELSE 0 END) / SUM(CASE WHEN `res`.`probe_2` IS NULL OR IFNULL(`res`.`calculated_score`, \'pass\') = \'fail\' THEN 0 ELSE 1 END) END')))

```

Comment:
This query also has a group by clause which means the case statement will evaluate the first calculated_score value in each group rather than each participant's calculated_score. The SUM aggregate function needs to wrap around all of the evaluations and calculations to get a true average CT value.

If your goal here is to deal with `S1EPT-10: Hide Field Testing Mean Ct Value When Expected Ct Value is 0`, then you need to account for lowest value between ref.ultra_probe_is1081_is6110, ref.ultra_probe_rpo_b1, ref.ultra_probe_rpo_b2, ref.ultra_probe_rpo_b3 & ref.ultra_probe_rpo_b4. If any of these values is 0, the red bar should not appear on the chart.

Lastly, you've made this change to the MTB/RIF ultra query only. This report has two sections; one for MTB/RIF submissions and one for MTB/RIF Ultra submissions. S1EPT-10 expects that both sections hide the mean submitted Ct value when the expected value is 0.

Reply:
I had to do this and push it to the live website because of an urgent request from the customer. For reference, see https://github.com/SystemOneId/ept/pull/13

Please revert this change?